CMakeLists.txt

```diff
@@ -128,6 +130,42 @@ list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_LIST_DIR}/third_party/abseil-cpp/absl/copts/
 )
 
+#-------------------------------------------------------------------------------
+# Cross compiling configuration
+#-------------------------------------------------------------------------------
+
+if(CMAKE_CROSSCOMPILING)
+  if(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
+    message(FATAL_ERROR "Cross compilation with Windows host system is not supported yet")
+  endif()
+
+  message(STATUS "Detected cross compilation mode; configuring IREE on host...")
+
+  # C/C++ compilers for host compilation.
+  # Note: we need to explicitly set this because IREE does not work well with
+  # GCC at the moment: https://github.com/google/iree/issues/1269
+  set(IREE_HOST_C_COMPILER "$ENV{IREE_HOST_C_COMPILER}" CACHE FILEPATH "C compiler for host compilation")
+  set(IREE_HOST_CXX_COMPILER "$ENV{IREE_HOST_CXX_COMPILER}" CACHE FILEPATH "C++ compiler for host compilation")
```

Comment:
The cross-compilation difficulty with CMake actually comes from it's cumbersome handling of multiple toolchains. :) When cross-compiling, `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER` will be set to the target platform's toolchain, but there we want the toolchain for host platform. So I think we cannot use `CMAKE_C_COMPIER` and `CMAKE_CXX_COMPILER` as the default here. But we might be able to do that inside `iree_create_configuration`.. if we can handle GCC nicer.  `iree_create_configuration` is set up to handle setting C/C++ compilers via `$IREE_${CONFIG_NAME}_C_COMPILER` and `$IREE_${CONFIG_NAME}_CXX_COMPILER`. So actually here we can just delete these variables and it should work fine (the user will still need to specify the host compilers explicitly via CLI options). I still want to have it here because it allows having environment variables and it allows us to explicitly point out there is a bug here preventing us to just default to `CMAKE_C_COMPIER`/`CMAKE_CXX_COMPILER` in `iree_create_configuration`.